modules/features/search/src/main/java/au/com/shiftyjelly/pocketcasts/search/SearchHandler.kt

```diff
@@ -101,50 +106,80 @@ class SearchHandler @Inject constructor(
         }
 
     private val subscribedPodcastUuids = podcastManager
-        .findSubscribedRxSingle()
+        .findSubscribedFlow()
+        .asObservable()
         .subscribeOn(Schedulers.io())
-        .toObservable()
         .map { podcasts -> podcasts.map(Podcast::uuid).toHashSet() }
 
-    private val autoCompleteResults = searchQuery
-        .subscribeOn(Schedulers.io())
-        .map { it.string.trim() }
-        .debounce(200L, TimeUnit.MILLISECONDS)
-        .switchMap { query ->
+    @OptIn(ExperimentalCoroutinesApi::class)
+    private val autoCompleteResults = searchQuery.filter { it is Query.Suggestions }.asFlow()
+        .map { it.term.trim() }
+        .flatMapLatest { query ->
             if (query.isEmpty()) {
-                Observable.just<AutoCompleteSearch>(AutoCompleteSearch.Success(searchTerm = query, results = emptyList()))
+                flowOf(
+                    SearchUiState.SearchOperation.Success(
+                        searchTerm = query,
+                        results = emptyList(),
+                    ),
+                )
             } else {
-                rxObservable<List<SearchAutoCompleteItem>> {
-                    autoCompleteManager.autoCompleteSearch(term = query)
-                }.subscribeOn(Schedulers.io())
-                    .map<AutoCompleteSearch> {
-                        AutoCompleteSearch.Success(
+                flowOf(autoCompleteManager.autoCompleteSearch(term = query))
```

Comment:
🚫 I suggested `flowOf()` only for the empty case because this can cause a crash now. Turn on the airplane mode and type in anything. It is because exception happens not while emitting but before a `Flow` is even created. This should be either changed back to `flow{}` or `autoCompleteManager.autoCompleteSearch(term = query)` should be surrounded with a `try/catch`.